modules/common/service/service.go

```diff
@@ -39,12 +41,50 @@ func NewService(
 	service *corev1.Service,
 	labels map[string]string,
 	timeout time.Duration,
-) *Service {
-	return &Service{
+	override *OverrideSpec,
```

Comment:
This requires follow up changes to ovn-operator, infra-operator (memcached, redis) and swift operator due to the change in NewService()